editoast/simulation/src/lib.rs

```diff
@@ -1,3 +1,341 @@
 mod envs;
 
+use std::iter;
+use std::sync::Arc;
+
+// Crate-level exports
 pub use envs::core::CoreEnv;
+pub use envs::pathfinding::PathfindingConsist;
+pub use envs::pathfinding::PathfindingConstraints;
+pub use envs::pathfinding::PathfindingEnv;
+
+use futures::stream;
+use itertools::Itertools as _;
+use itertools::izip;
+use serde::de::DeserializeOwned;
+use serde::ser::Serialize;
+use tracing::Instrument;
+
+/// Indicates that a simulation can be performed and cached
+///
+/// This trait is meant to be implemented for simulation requests.
+///
+/// Features:
+/// - [fn Simulation::simulate] that performs a single simulation
+/// - [trait SimulationStreamExt] to batch simulations (requires additional [type Simulation::Context] bounds)
+pub trait Simulation: Sized + Send {
+    /// Simulation output
+    type Output: DeserializeOwned + Serialize + Send;
+    /// Computation error when performing the simulation for cache misses
+    type Error: std::error::Error + Send;
+    /// Context required for simulation
+    ///
+    /// Additional bounds are required on this to batch simulation requests, cf. [trait SimulationStreamExt].
+    type Context: Send;
+
+    /// Number of cache entry read attempts per batch
+    ///
+    /// Customize this value based on the size of simulation results considering all reads hit.
+    const CACHE_READS_BATCH_SIZE: usize = 50; // 🫳🎩
```

Comment:
I like the emoji, but I would add some instructions for the next developer, something along the lines of:

```suggestion
    const CACHE_READS_BATCH_SIZE: usize = 50; // This value has been chosen this way🫳🎩: please adjust if you have more educated information (and adjust the comment 😉)
```

Reply:
Made specifying the constant necessary for each implementor so that everyone has to think about this issue [4fb5625ed6](https://github.com/OpenRailAssociation/osrd/pull/13642/commits/4fb5625ed6e7a0c6a56f273cfafe800a8bca82d8).